pkcs11/src/new/types/mod.rs

```diff
@@ -337,3 +291,75 @@ impl From<Flags> for pkcs11_sys::CK_FLAGS {
         flags.flags
     }
 }
+
+#[repr(u8)]
+#[derive(Debug)]
+pub enum Bbool {
+    False = 0,
+    True = 1,
+}
+
+impl TryFrom<&[u8]> for Bbool {
+    type Error = Error;
+
+    fn try_from(slice: &[u8]) -> Result<Self> {
+        CK_BBOOL::from_ne_bytes(slice.try_into()?).try_into()
+    }
+}
+
+impl From<Bbool> for CK_BBOOL {
+    fn from(bbool: Bbool) -> Self {
+        bbool as CK_BBOOL
+    }
+}
+
+impl From<bool> for Bbool {
+    fn from(val: bool) -> Self {
+        if val {
+            Bbool::True
+        } else {
+            Bbool::False
+        }
+    }
+}
+
+impl TryFrom<CK_BBOOL> for Bbool {
+    type Error = Error;
+
+    fn try_from(bbool: CK_BBOOL) -> Result<Self> {
+        match bbool {
+            CK_FALSE => Ok(Bbool::False),
+            CK_TRUE => Ok(Bbool::True),
+            other => {
+                error!("Bbool value {} is not supported.", other);
+                Err(Error::NotSupported)
```

Comment:
Hmm, I'd say `NotSupported` isn't the right return value here - it kinda implies it _could_ be supported, but not yet. Is there something more like `InvalidValue`?

Reply:
Ah yes true, will add that error!